open-api/rest-catalog-open-api.yaml

```diff
@@ -1795,6 +1795,61 @@ paths:
         5XX:
           $ref: '#/components/responses/ServerErrorResponse'
 
+  /v1/{prefix}/events:
+    parameters:
+      - $ref: '#/components/parameters/prefix'
+    post:
+      tags:
+        - Catalog API
+      summary: Get events for changes to catalog objects
+      description: >
+        Returns a sequence of changes to catalog objects (tables, namespaces, views) 
+        that allows clients to efficiently track metadata modifications without polling individual 
+        resources. Consumers track their progress through a next-page-token, enabling resumable 
+        synchronization after downtime or errors.
+
+        This endpoint primarily supports use cases like catalog federation, workflow triggering, 
+        and basic audit capabilities.
+
+        When a request includes a `next-page-token`, it takes precedence over 
+        all other filters including `after-timestamp-ms` and `operation-type`.
+        The server encodes all necessary state in the token to ensure 
+        consistent filtering across pages. Clients should use the returned `next-page-token` for 
+        subsequent requests.
+
+        Consumers should be prepared to handle 410 Gone responses when requested sequences are
+        outside the server's retention window. Consumers should also de-duplicate received events based
+        on the event's `event-id`. Consistency guarantees vary between server implementations.
+      operationId: getEvents
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/QueryEventsRequest'
+        required: true
+      responses:
+        200:
+          description: A sequence of change events to catalog objects
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/EventsResponse'
```

Comment:
nit: Maybe QueryEventsResponse to be inline with the name of the request?

Reply:
Good catch, name changed in latest commit.